File: 2. 구현/문자열 압축.py
def solution(s):
    answer = len(s)

    for step in range(1, len(s) // 2 + 1):
      prev = s[:step]
      compressed = ''
      cnt = 1
      for i in range(step, len(s), step):
        if prev == s[i : i + step]:
          cnt += 1
        else:
          compressed += str(cnt) + prev if cnt > 1 else prev
          prev = s[i : i + step]
          cnt = 1
      compressed += str(cnt) + prev if cnt > 1 else prev
      answer = min(answer, len(compressed))

    return answer

File: codeEg/2. 구현/문자열 압축.js
function solution(s) {
  let answer = s.length

  for (let step = 1; step < parseInt(s.length / 2) + 1; step++) {
    let compressed = ''
    let cnt = 1
    let prev = s.slice(0, step)
    for (let i = step; i < s.length; i += step) {
      if (prev === s.slice(i, i + step)) {
        cnt += 1
      } else {
        compressed += cnt > 1 ? cnt.toString() + prev : prev
        prev = s.slice(i, i + step)
        cnt = 1
      }
    }
    compressed += cnt > 1 ? cnt.toString() + prev : prev
    answer = Math.min(answer, compressed.length)
  }
  return answer
}